src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs

```diff
@@ -260,6 +263,7 @@ public static bool ReferenceMatchesDefinition(AssemblyName? reference, AssemblyN
             return refName.Equals(defName, StringComparison.OrdinalIgnoreCase);
         }
 
+        [RequiresAssemblyFiles(Message = "The code will throw for assemblies embedded in a single-file app")]
```

Comment:
It will? where?

Reply:
Yikes, I thought the documentation said something about it, but it was [CodeBase ](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.codebase?view=net-5.0) the one that throws. I will change the message to reflect string.Empty instead.

Reply:
I created a program to verify the behavior
```
var a = Assembly.GetExecutingAssembly().GetName();
_ = a.CodeBase;  // returns string.Empty
_ = a.EscapedCodeBase;  // returns string.Empty
var b = Assembly.GetExecutingAssembly();
_ = b.CodeBase;  // throws and generates obsolete warning
_ = b.EscapedCodeBase; // throws and generates obsolete warning
```
It depends on if you call CodeBase/EscapedCodeBase from AssemblyName.cs or Assembly.cs the behavior it will take. 
Calling EscapedCodeBase from assembly will generate the following stack trace
```
Unhandled exception. System.NotSupportedException: CodeBase is not supported on assemblies loaded from a single-file bundle.
   at System.Reflection.RuntimeAssembly.get_CodeBase()
   at System.Reflection.Assembly.get_EscapedCodeBase()
   at EscapedCodeBase.Program.Main(String[] args)
```
Here we can see that uses the [RuntimeAssembly ](https://github.com/dotnet/runtime/blob/main/src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs#L94) implementation which throws.
In short, I need to modify the messages to reflect the behavior